bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/discovery/OpenSprinklerDiscoveryService.java

```diff
@@ -76,78 +72,71 @@ OpenSprinklerApiFactory getApiFactory() {
+                .withLabel("OpenSprinkler HTTP Bridge").withRepresentationProperty("hostname").build());
+        // Now create the Device thing
+        properties.clear();
+        properties.put("hostname", ip);
+        ThingUID uid = new ThingUID(OPENSPRINKLER_DEVICE, bridgeUID, ip.replace('.', '_'));
+        thingDiscovered(DiscoveryResultBuilder.create(uid).withBridge(bridgeUID).withProperties(properties)
+                .withRepresentationProperty("hostname").withLabel("OpenSprinkler Device").build());
     }
 
-    /**
-     * Provide a string list of all the IP addresses associated with the network interfaces on
-     * this machine.
-     *
-     * @return String list of IP addresses.
-     * @throws UnknownHostException
-     * @throws SocketException
-     */
-    private List<String> getIpAddressScanList() throws UnknownHostException, SocketException {
-        List<String> results = new ArrayList<>();
-
-        InetAddress localHost = InetAddress.getLocalHost();
-        NetworkInterface networkInterface = NetworkInterface.getByInetAddress(localHost);
-
-        for (InterfaceAddress address : networkInterface.getInterfaceAddresses()) {
-            InetAddress ipAddress = address.getAddress();
-
-            String cidrSubnet = ipAddress.getHostAddress() + "/" + address.getNetworkPrefixLength();
-
-            /* Apache Subnet Utils only supports IP v4 for creating string list of IP's */
-            if (ipAddress instanceof Inet4Address) {
-                logger.debug("Found interface IPv4 address to scan: {}", cidrSubnet);
-
-                SubnetUtils utils = new SubnetUtils(cidrSubnet);
-
-                results.addAll(Arrays.asList(utils.getInfo().getAllAddresses()));
-            } else if (ipAddress instanceof Inet6Address) {
-                logger.debug("Found interface IPv6 address to scan: {}", cidrSubnet);
-            } else {
-                logger.debug("Found interface unknown IP type address to scan: {}", cidrSubnet);
+    private void ipAddressScan() {
+        String ipAddress;
+        try {
+            for (Enumeration<NetworkInterface> enumNetworks = NetworkInterface.getNetworkInterfaces(); enumNetworks
+                    .hasMoreElements();) {
+                NetworkInterface networkInterface = enumNetworks.nextElement();
+                for (Enumeration<InetAddress> enumIpAddr = networkInterface.getInetAddresses(); enumIpAddr
+                        .hasMoreElements();) {
+                    InetAddress inetAddress = enumIpAddr.nextElement();
+                    if (!inetAddress.isLoopbackAddress() && inetAddress.isSiteLocalAddress()) {
+                        ipAddress = inetAddress.getHostAddress().toString();
+                        String temp = ipAddress.substring(0, ipAddress.lastIndexOf("."));
+                        for (int i = 1; i < 254; i++) {
+                            String host = temp + "." + i;
```

Comment:
This will only work if the network mask is /24.

Reply:
FIXED: added more code to handle different subnets.